### ISSUE TITLE
chainer vs. pytorch in terms of training speed

### ISSUE THREAD

**Issue by sw005320** [espnet/espnet#28]
Can you summarize chainer vs. pytorch back ends in terms of the training time?
With @ShigekiKarita 's efforts, we can compare them with almost same conditions (maybe with blstmp? @ShigekiKarita do you still need some time to finish vggblstmp in pytorch?). I will report them to the chainer issue. The chainer developer would give us some solutions.

**Comment by ShigekiKarita** [comment]
these results are obtained with the final `test-pytorch-loss` branch (merged in #20) by Tesla K80 (not 1080ti)


## WSJ training results of blstmp

| |chainer| pytorch|
|:--|----:|----:|
|valid acc|93.6|93.9|
|dev CER|10.7|11.1|
|test CER|8.3|8.2|
|time|82129|34496|


additional info
- chainer https://github.com/espnet/espnet/issues/9#issuecomment-352322057 and [chainer.log](https://github.com/espnet/espnet/files/1585191/chainer.log)
- pytorch https://github.com/espnet/espnet/issues/9#issuecomment-353284921 and [pytorch.log](https://github.com/espnet/espnet/files/1585189/pytorch.log) 




## WSJ training results of vggblstmp

Here are vggblstmp training results in chainer and pytorch.

- chainer
![acc](https://user-images.githubusercontent.com/6745326/34330560-8b25915c-e961-11e7-9b28-b26bd747f3da.png)

- pytorch
![acc](https://user-images.githubusercontent.com/6745326/34330568-9f4d92e2-e961-11e7-9f92-35716ff17a9b.png)

As seen in these figures, we still have something wrong with pytorch vggblstmp. I'm adding some equality checks for VGG2L.

**Comment by kan-bayashi** [comment]
I have no result of vggblsmtp with pytorch.
So I show the result of vggblstmp with chainer and that of blstmp with pytorch.

## tedlium recipe
|   | Chainer VGGMBLSTMP | Pytorch BLSTMP |
----|----|---- 
| dev CER | 12.3 % | 12.6 % |
| test CER | 12.0 % | 12.4 % |
| time (titan x pascal) | 339424 sec (94.3 h)  | 50686 sec (14.1 h) |

## csj recipe
|   | Chainer VGGMBLSTMP | Pytorch BLSTMP |
----|----|---- 
| eval1 CER | 11,3 % |  % |
| eval2 CER | 7.8 % |  % |
| eval2 CER | 9.2 % |  % |
| time (titan x pascal) | 493666 sec (137.1 h)  | 164119 sec (45.6 h) |

## environmet
- chainer==3.2.0 installed by pip
- pytorch==0.3.0.post4 installed by pip
- Python==2.7.5
- CUDA 8.0
- CuDNN 6.0 
- GCC 4.8.5
- CentOS 7

**Comment by kan-bayashi** [comment]
Continue to discuss in #380
